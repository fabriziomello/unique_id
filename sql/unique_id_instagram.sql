\set VERBOSITY terse
CREATE EXTENSION unique_id;

-- Throw error because should exists a sequence
SELECT unique_id_instagram('instagram_seq');

CREATE SEQUENCE instagram_seq;

-- Individual transactions and default shard_id = 0
SELECT unique_id_instagram('instagram_seq') > unique_id_instagram('instagram_seq');
SELECT unique_id_instagram('instagram_seq') < unique_id_instagram('instagram_seq');
SELECT unique_id_instagram('instagram_seq') <> unique_id_instagram('instagram_seq');

-- Individual transactions and shard_id = 1
SELECT unique_id_instagram('instagram_seq', 1) > unique_id_instagram('instagram_seq', 1);
SELECT unique_id_instagram('instagram_seq', 1) < unique_id_instagram('instagram_seq', 1);
SELECT unique_id_instagram('instagram_seq', 1) <> unique_id_instagram('instagram_seq', 1);

DROP EXTENSION unique_id;
