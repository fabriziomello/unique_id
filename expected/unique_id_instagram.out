\set VERBOSITY terse
CREATE EXTENSION unique_id;
-- Throw error because should exists a sequence
SELECT unique_id_instagram('instagram_seq');
ERROR:  relation "instagram_seq" does not exist at character 28
CREATE SEQUENCE instagram_seq;
-- Individual transactions and default shard_id = 0
SELECT unique_id_instagram('instagram_seq') > unique_id_instagram('instagram_seq');
 ?column? 
----------
 f
(1 row)

SELECT unique_id_instagram('instagram_seq') < unique_id_instagram('instagram_seq');
 ?column? 
----------
 t
(1 row)

SELECT unique_id_instagram('instagram_seq') <> unique_id_instagram('instagram_seq');
 ?column? 
----------
 t
(1 row)

-- Individual transactions and shard_id = 1
SELECT unique_id_instagram('instagram_seq', 1) > unique_id_instagram('instagram_seq', 1);
 ?column? 
----------
 f
(1 row)

SELECT unique_id_instagram('instagram_seq', 1) < unique_id_instagram('instagram_seq', 1);
 ?column? 
----------
 t
(1 row)

SELECT unique_id_instagram('instagram_seq', 1) <> unique_id_instagram('instagram_seq', 1);
 ?column? 
----------
 t
(1 row)

DROP EXTENSION unique_id;
